src/libdmg/con_manager.h

```diff
@@ -0,0 +1,81 @@
+//
+// Created by mattcoding4days on 3/24/21.
+// It is a good idea to separate all DMG's source code into its own folder, essentially
+// acting as a header only library, at least for the TransactionFilter class.
+//
+
+#ifndef BLOCKSEER_LISTENER_BTC_NODE_CON_MANAGER_H
+#define BLOCKSEER_LISTENER_BTC_NODE_CON_MANAGER_H
+
+// standard lib
+#include <string>
+#include <utility>
+#include <memory>
+
+// sql library
+#include <mysql_connection.h>
+#include <mysql_driver.h>
+#include <cppconn/exception.h>
+#include <cppconn/resultset.h>
+#include <cppconn/statement.h>
+
+
+namespace libDMG
+{
+    /* NOTE: _ConnectionDetails should not be used directly
+     * */
+    struct ConnectionDetails
+    {
+    protected:
+        std::string m_host;
+        std::string m_user;
+        std::string m_pass;
+        std::string m_database_name;
+
+    public:
+        ConnectionDetails();
+    };
+
+
+    /* NOTE: ConnectionManager only handles connections to mysql and executes queries,
+     * it is not meant to have any other functionality. Encapsulating the sql Library functionality
+     * in this object is crucial, since MySQL_Driver is a Singleton, it is very rigid to work with.
+     * sql::Connection also has a private copy constructor, so passing it around is very difficult as only
+     * move semantics can be utilized. A safer approach would have been to make unique_ptr objects out of
+     * Driver and Connection, but that approach lead to far more code, and was becoming un-managable. In the name
+     * of simplicity I chose the regular ol' pointer approach. If the code fails here for any reason, it is a possibillity
+     * the destructor won't be called. But the bitcoin StartShutdown() function should solve that problem.
+     *
+     * NOTE: since MySQL_Driver is a singleton it deletes it's own memory, the destructor is private and thus cannot be called.
+     * */
+    class ConnectionManager : public ConnectionDetails
+    {
+    private:
+        sql::mysql::MySQL_Driver *m_driver;
+    protected:
```

Comment:
Since we don't have a coding standard yet, this is up for debate: My opinion would be that until we have a reason to inherit, we make everything private.

Reply:
fair enough, will change all members to private